src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs

```diff
@@ -0,0 +1,416 @@
+            {
+                var indentText = indentation.Offset.CreateIndentationString(documentOptions.GetOption(FormattingOptions.UseTabs), documentOptions.GetOption(FormattingOptions.TabSize));
+                var indentTextChange = new TextChange(new TextSpan(indentation.BasePosition, 0), indentText);
+                return originalText.WithChanges(indentTextChange);
+            }
+        }
+
+        public override async Task<bool> IsValidForBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken)
+        {
+            // Only potentially valid for curly brace completion if not in an interpolation brace completion context.
+            if (OpeningBrace == brace && await InterpolationBraceCompletionService.IsPositionInInterpolationContextAsync(document, openingPosition, cancellationToken).ConfigureAwait(false))
+            {
+                return false;
+            }
+
+            return await base.IsValidForBraceCompletionAsync(brace, openingPosition, document, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected override bool IsValidOpeningBraceToken(SyntaxToken token) => token.IsKind(SyntaxKind.OpenBraceToken) && !token.Parent.IsKind(SyntaxKind.Interpolation);
+
+        protected override bool IsValidClosingBraceToken(SyntaxToken token) => token.IsKind(SyntaxKind.CloseBraceToken);
+
+        private bool ContainsOnlyWhitespace(SourceText text, int openingPosition, int closingPosition)
+        {
+            var start = openingPosition;
+            start = text[start] == OpeningBrace ? start + 1 : start;
+
+            var end = closingPosition - 1;
+            end = text[end] == ClosingBrace ? end - 1 : end;
+
+            if (!PositionInSnapshot(start, text) ||
+                !PositionInSnapshot(end, text))
+            {
+                return false;
+            }
+
+            for (var i = start; i <= end; i++)
+            {
+                if (!char.IsWhiteSpace(text[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+
+            static int GetValueInValidRange(int value, int smallest, int largest)
+            => Math.Max(smallest, Math.Min(value, largest));
+
+            static bool PositionInSnapshot(int position, SourceText text)
+                => GetValueInValidRange(position, 0, Math.Max(0, text.Length - 1)) == position;
+        }
+
+        /// <summary>
+        /// Formats the span between the opening and closing points, options permitting.
+        /// Returns the text changes that should be applied to the input document to 
+        /// get the formatted text and the new closing point in the formatted text.
+        /// </summary>
+        private static async Task<(ImmutableArray<TextChange> TextChanges, int NewClosingPoint)> FormatTrackingSpanAsync(
```

Comment:
```suggestion
        private static async Task<(ImmutableArray<TextChange> textChanges, int newClosingPoint)> FormatTrackingSpanAsync(
```

we do not capitalize tuple members.